ssz/codec.py

```diff
@@ -13,14 +14,26 @@
 )
 
 
-def encode(value, sedes=None):
+def encode(value, sedes=None, cache=True):
     """
     Encode object in SSZ format.
     `sedes` needs to be explicitly mentioned for encode/decode
     of integers(as of now).
     `sedes` parameter could be given as a string or as the
     actual sedes object itself.
     """
+    if isinstance(value, Serializable):
+        cached_ssz = value._cached_ssz
+        if sedes is None and cached_ssz:
```

Comment:
```suggestion
        if sedes is None and cached_ssz is not None:
```

just to explicitly avoid the case `cached_ssz == b""`

Reply:
Why do we check for `sedes is None` here?

Reply:
Added some comment (copied from pyrlp): https://github.com/ethereum/py-ssz/pull/37/commits/d166d6c96deb080c73c2d13f7389bdd34c7bc8f1

Does that make sense to py-ssz as well? `infer_sedes` part seems similar.